hazelcast/src/main/java/com/hazelcast/concurrent/semaphore/SemaphoreService.java

```diff
@@ -139,7 +139,7 @@ public void beforeMigration(PartitionMigrationEvent partitionMigrationEvent) {
 
     @Override
     public Operation prepareReplicationOperation(PartitionReplicationEvent event) {
-        Map<String, SemaphoreContainer> migrationData = new HashMap<String, SemaphoreContainer>();
+        Map<String, SemaphoreContainer> migrationData = createHashMap(containers.size() / 2);
```

Comment:
Reverting this as it assumes half of the containers will be replicated which does not need to be true.